src/dcd/common/constants.d

```diff
@@ -26,87 +26,122 @@ struct ConstantCompletion
 	string ddoc;
 }
 
+private static immutable pragmaDDoc = import("pragma.dd");
+private static immutable traitsDDoc = import("traits.dd");
+
+template fetchDocDDforDT(string dtStart)
+{
+	enum fetchDocDDforDT = {
+		import std.string : lineSplitter, KeepTerminator, stripLeft, stripRight,
+			strip, startsWith, endsWith;
+
+		bool found;
+		string ret;
+		string indent;
+		bool inCode;
+		foreach (line; pragmaDDoc.lineSplitter!(KeepTerminator.yes))
+		{
+			if (line.stripLeft.startsWith("$(DT $(LNAME2 ")
+					|| line.stripLeft.startsWith("$(SPEC_SUBNAV_PREV_NEXT"))
+			{
+				if (found) // abort on new section if in section
+					break;
+				else if (line.stripLeft.startsWith("$(DT $(LNAME2 " ~ dtStart)) // set section if correct
+				{
+					indent = line[0 .. $ - line.stripLeft.length];
+					found = true;
+				}
+			}
+			else if (found)
+			{
+				if (line.startsWith("---")) // code blocks aren't indented
+					inCode = !inCode;
+
+				if (inCode)
+					ret ~= line;
+				else
+				{
+					if (line.startsWith(indent)) // strip indentation of DT
+						ret ~= line[indent.length .. $];
+					else
+						ret ~= line;
+				}
+			}
+		}
+		if (!found)
+			throw new Exception("DT '" ~ dtStart ~ "' was not found");
+		// ret still has `$(DD content)` around it, strip that
+		if (ret.startsWith("$(DD"))
+		{
+			ret = ret[4 .. $].strip;
+			if (ret.endsWith(")"))
+				ret = ret[0 .. $ - 1].stripRight;
+		}
+		return ret;
+	}();
+}
+
+template fetchDocByGNAME(string gname)
```

Comment:
I don't see a reason why this warrants a special function.

Reply:
hm the two functions are specialized for their files right now, not really usable in anything with different formatting except more or less indentation, so both get a different implementation